transport/socks5/packet_listener_test.go

```diff
@@ -0,0 +1,107 @@
+package socks5
+
+import (
+	"bytes"
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/things-go/go-socks5"
+)
+
+func TestSOCKS5Associate(t *testing.T) {
+	// Create a local listener
+	// This creates a UDP server that responded to "ping"
+	// message with "pong" response
+	locIP := net.ParseIP("127.0.0.1")
+	// Create a local listener
+	echoServerAddr := &net.UDPAddr{IP: locIP, Port: 12199}
```

Comment:
No hard-coded ports for local servers. Retrieve it from the listener local address instead.